ionic2/src/components/filter-pokemon-tab/filter-pokemon-tab.component.ts

```diff
@@ -29,7 +29,9 @@ export class FilterPokemonTabComponent implements OnInit {
     pokemonTypes: []
   };
 
-  constructor(private apiService: ApiService, private filterService: FilterService) { }
+  constructor(private apiService: ApiService,
+              private filterService: FilterService,
+              private typeService: TypeService) { }
```

Comment:
!For future reference only, i'll fix it asap.
I missed something there: This should be public.
Reason being: If you use the service in a template, the AoT compile must be able to access it.